route/result/download.js

```diff
@@ -61,12 +61,14 @@ function route (app) {
     app.express.get('/:id/:rid.csv', getTaskAndResult, function (req, res) {
         var task = res.locals.task;
         var result = res.locals.result;
-        var rows = ['"code","message","type"'];
+        var rows = ['"code","message","type","context","selector"'];
         result.results.forEach(function (msg) {
             rows.push([
                 JSON.stringify(msg.code),
                 JSON.stringify(msg.message),
-                JSON.stringify(msg.type)
+                JSON.stringify(msg.type),
+                JSON.stringify(msg.context),
+                JSON.stringify(msg.selector)
```

Comment:
It might be nice to make sure these are strings so that they're quoted in the output. We currently have a large set of results with no context or selector so they will be `undefined`. Just something like this would do:

```js
JSON.stringify(msg.context || ''),
JSON.stringify(msg.selector || '')
```